packages/graphile-build-pg/src/plugins/viaTemporaryTable.js

```diff
@@ -101,16 +115,38 @@ export default async function viaTemporaryTable(
     const rawValues = rows.map(row => row && row[firstKey]);
     const values = rawValues.filter(rawValue => rawValue !== null);
     const convertFieldBack = isPgClassLike
-      ? sql.query`(str::${sqlTypeIdentifier}).*`
-      : sql.query`str::${sqlTypeIdentifier} as ${sqlResultSourceAlias}`;
+      ? sql.query`\
+              select (str::${sqlTypeIdentifier}).*
+              from unnest((${sql.value(values)})::text[]) str`
+      : isPgRecord
+        ? sql.query`\
+          select ${sql.join(
+            outputArgNames.map(
+              (outputArgName, idx) =>
+                sql.query`\
+                    (arr)[${sql.literal(idx + 1)}]::${sql.identifier(
+                  outputArgTypes[idx].namespaceName,
+                  outputArgTypes[idx].name
+                )} as ${sql.identifier(
+                  outputArgName !== "" ? outputArgName : `column${idx + 1}`
+                )}`
+            ),
+            ", "
+          )}
+          from (values ${sql.join(
+            values.map(value => sql.query`(${sql.value(value)}::text[])`),
+            ", "
+          )}) as x(arr)`
```

Comment:
Turns out using a values list is exactly what you did 👍

Reply:
I'm wondering if `jsonb_to_recordset` is safe to use for *just* `isPgRecord` (since it's new behaviour so definitely not breaking) which'd avoid this complexity.